e2e/commands_test.go

```diff
@@ -316,3 +317,40 @@ func testFork(registry string) func(*testing.T) {
 		assert.Assert(t, golden.Bytes(metadata2, "expected-fork-metadata-no-rename.golden"))
 	}
 }
+
+func TestExternalFilesWithRegistry(t *testing.T) {
+	r := startRegistry(t)
+	defer r.Stop(t)
+	registry := r.GetAddress(t)
+	t.Run("externalFiles", testExternalFiles(registry))
+}
+func testExternalFiles(registry string) func(*testing.T) {
+	return func(t *testing.T) {
+		dir := fs.NewDir(t, "testexternalfiles",
+			fs.WithDir("externalfiles.dockerapp", fs.FromDir("testdata/externalfiles.dockerapp")),
+		)
+		defer dir.Remove()
+
+		icmd.RunCommand(dockerApp, "push", "--namespace", registry+"/acmecorp", dir.Join("externalfiles.dockerapp")).Assert(t, icmd.Success)
+		// inspect will run the core pull code too
+		result := icmd.RunCommand(dockerApp, "inspect", registry+"/acmecorp/externalfiles.dockerapp:0.1.0")
+		result.Assert(t, icmd.Success)
+		assert.Assert(t, strings.Contains(result.Combined(), "config.cfg"))
+		nestedpath := filepath.Join("nesteddir", "config2.cfg")
+		assert.Assert(t, strings.Contains(result.Combined(), nestedpath))
+		deepnestedpath := filepath.Join("nesteddir", "nested2", "nested3", "config3.cfg")
```

Comment:
I think `filepath.Join` shouldn't be used here. It means the `docker inspect` output changes depending the os, so on windows it will print `nesteddir\nested2\nested3\config3.cfg`, but what about having the same has linux? `nesteddir/nested2/nested3/config3.cfg` ?